client-tez/src/main/java/org/apache/tez/dag/app/TezRemoteShuffleManager.java

```diff
@@ -305,13 +306,19 @@ public ShuffleAssignmentsInfo run() throws Exception {
                                           RssTezConfig.toRssConf(conf)
                                               .get(MAX_CONCURRENCY_PER_PARTITION_TO_WRITE),
                                           0,
-                                          keyClassName,
-                                          valueClassName,
-                                          comparatorClassName,
-                                          conf.getInt(
-                                              RssTezConfig.RSS_MERGED_BLOCK_SZIE,
-                                              RssTezConfig.RSS_MERGED_BLOCK_SZIE_DEFAULT),
-                                          conf.get(RssTezConfig.RSS_REMOTE_MERGE_CLASS_LOADER)));
+                                          RssProtos.MergeContext.newBuilder()
```

Comment:
For tez, RssOrderedPartitionedKVOutput will trigger AppMaster to registerShuffle. If merge is disable, the parameters are null. Otherwise, these are not null.
So I did not check whether remote merge is enable. Here I think we should check keyClass, if null, we will pass null, no need to construct MergeContext.
By this way, the code logic is clearer.

Reply:
Good suggestion